lewis_emulators/wbvalve/interfaces/stream_interface.py

```diff
@@ -0,0 +1,47 @@
+from lewis.adapters.stream import StreamInterface, Cmd
+from lewis.core.logging import has_log
+
+from lewis_emulators.utils.command_builder import CmdBuilder
+from lewis_emulators.utils.replies import conditional_reply
+
+if_connected = conditional_reply("connected")
+
+@has_log
+class WbvalveStreamInterface(StreamInterface):
+
+    in_terminator = '\n'
+    out_terminator = '\n'
+
+    def __init__(self):
+
+        super(WbvalveStreamInterface, self).__init__()
+        self.commands = {
+            CmdBuilder(self.get_position).escape("st").build(),
+            CmdBuilder(self.set_position).escape("wb").int().escape("on").build()
+        }
```

Comment:
This works ok but in general you shouldn't need to override the `__init__` function here, you can just define `commands` as a dictionary at class level and refer to the functions by name (unless you're doing something particularly clever)

Reply:
It is nicer to refer to function names by reference because then refactoring and ctrl-click works on them without issue. This can only be done in init.